### ISSUE TITLE
Registering custom picklers for a tree of classes

### ISSUE THREAD

**Issue by frensjan** [irmen/Pyrolite#35]
Hi, like working with the pickle part of Pyrolite for Apache Spark / Cassandra / Python integration. However, as I'm using / touching Scala / Cassandra code I need to add some custom picklers and constructors. That's obvious of course, but things would be a lot easier and above all, less error prone if I could register a pickler for an entire tree of classes at once by registering it for a parent class / interface.

To illustrate the issue: in order to support pickling Scala maps, it's easy to write a single simple implementation of ```net.razorvine.pickle.IObjectPickler```. However it has to be registered to every possible specialization of ```scala.collection.Map```. In practice that means registering it for ```scala.collection.immutable.Map.Map1``` and ```scala.collection.immutable.Map.Map2```, etc.

Another example is ```java.nio.ByteBuffer```. The actual objects you'd probably see flying around are of the type ```java.nio.HeapByteBuffer```, which is a package protected class. So you'd have to register pickler something like:
```scala
Pickler.registerCustomPickler(Class.forName("java.nio.HeapByteBuffer"), ByteBufferPickler)
```
while
```scala
Pickler.registerCustomPickler(classOf[ByteBuffer], ByteBufferPickler)
```
should capture just any specialization of ```ByteBuffer```.

Cheers!

**Comment by irmen** [comment]
I can see the value of such a change.  Should perhaps also be done for the Serpent serializer if this finds its way in the Pickle serializer.

**Comment by irmen** [comment]
Would it be sufficient if the distinction was made on the basis of registering an interface (to capture all implementing classes) and a specific class (to capture just that one specific class)?

**Comment by frensjan** [comment]
I would think so, yes! Perhaps there are cases where you'd want to catch an inheritance tree based on a root class ... might be an issue.

**Comment by irmen** [comment]
Just released new version of Serpent as well (1.13) with this feature corrected for Java and .NET

**Comment by frensjan** [comment]
nice!
